src/games/strategy/triplea/ai/proAI/util/ProOddsCalculator.java

```diff
@@ -0,0 +1,173 @@
+package games.strategy.triplea.ai.proAI.util;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.PlayerID;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+import games.strategy.triplea.Properties;
+import games.strategy.triplea.ai.proAI.ProData;
+import games.strategy.triplea.ai.proAI.data.ProBattleResult;
+import games.strategy.triplea.delegate.BattleCalculator;
+import games.strategy.triplea.delegate.Matches;
+import games.strategy.triplea.delegate.TerritoryEffectHelper;
+import games.strategy.triplea.oddsCalculator.ta.AggregateResults;
+import games.strategy.triplea.oddsCalculator.ta.IOddsCalculator;
+import games.strategy.util.Match;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Pro AI odds calculator.
+ */
+public class ProOddsCalculator {
+
+  private final IOddsCalculator calc;
+  private boolean isCanceled = false;
+
+  public ProOddsCalculator(final IOddsCalculator calc) {
+    this.calc = calc;
+  }
+
+  public void setData(final GameData data) {
+    calc.setGameData(data);
+  }
+
+  public void cancelCalcs() {
+    calc.cancel();
+    isCanceled = true;
+  }
+
+  public ProBattleResult estimateAttackBattleResults(final PlayerID player, final Territory t,
+      final List<Unit> attackingUnits, final List<Unit> defendingUnits, final Set<Unit> bombardingUnits) {
+
+    final ProBattleResult result = checkIfNoAttackersOrDefenders(t, attackingUnits, defendingUnits);
+    if (result != null) {
+      return result;
+    }
+
+    // Determine if attackers have no chance
+    final double strengthDifference = ProBattleUtils.estimateStrengthDifference(t, attackingUnits, defendingUnits);
+    if (strengthDifference < 45) {
+      return new ProBattleResult(0, -999, false, new ArrayList<Unit>(), defendingUnits, 1);
```

Comment:
Dude, I reject this argument. If you wanted a minimial fix, then this should have been a minimal fix. Moving stuff around this much, you are refactoring, and you did not go very far. It won't take more than 20 minutes to clean this class up. Then it'll be done once and for all.